ledger/store/src/helpers/rocksdb/internal/mod.rs

```diff
@@ -189,6 +193,26 @@ impl Database for RocksDB {
 }
 
 impl RocksDB {
+    /// Toggles the atomic override; if it becomes disabled, the pending
+    /// operations get executed.
+    fn flip_atomic_override(&self) -> Result<bool> {
+        // https://github.com/rust-lang/rust/issues/98485
+        let previous_value = self.atomic_override.load(Ordering::SeqCst);
+
+        // A flip from enabled to disabled executes all pending operations
+        // as a single atomic batch.
+        if previous_value {
+            let batch = mem::take(&mut *self.atomic_batch.lock());
+            self.rocksdb.write(batch)?;
```

Comment:
This will write to DB even if the `previous_atomic_depth` is not 1. Is this safe?

Normally in `finish_atomic` we only write if we're at depth 0  `if previous_atomic_depth == 1`. But now we can get around that by abusing `flip_atomic_override`. In theory if we use the override flag correctly there will be no issues, but it is a bit concerning that this is a possibility.

Reply:
I think it's reasonable to add an atomic depth safeguard here :+1:; it will restrict the applicability of the override, but we only want it used in one spot where this is a given.